### PR TITLE
datagrid in sms section

### PR DESCRIPTION
**Pull request by virgotopia**

### CODE REVIEW

File: client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ angular.module("managerApp").controller("TelecomSmsSmsTemplatesCtrl", class Tele
                 serviceName: this.$stateParams.serviceName
             }).$promise.then((templates) => {
                 this.templates.raw = templates;
+                for (let model of this.templates.raw) {
```
Comment (antleblanc) [inline client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js]: We should avoid using this syntax for the moment. Not yet covered by our babel config/target.

File: client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js [hunk 2]
```diff
@@ -78,12 +83,18 @@ angular.module("managerApp").controller("TelecomSmsSmsTemplatesCtrl", class Tele
...
         this.api.sms.templates.resetCache();
         this.api.sms.templates.resetQueryCache();
+        this.templates.paginated = [];
         this.templates.raw = null;
         this.templates.isLoading = true;
         return this.api.sms.templates.query({
             serviceName: this.$stateParams.serviceName
         }).$promise.then((templates) => {
             this.templates.raw = templates;
+            for (let model of this.templates.raw) {
```
Comment (antleblanc) [inline client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js]: See my previous comment.

File: client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js (hunk 1 above)
Comment (neolitec) [reply client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js]: for .. of is not in preset-es2015?

File: client/app/telecom/sms/options/response/telecom-sms-options-response.html [hunk 3]
```diff
@@ -175,99 +175,41 @@ <h1 data-translate="sms_options_response"></h1>
...
-                                <td class="text-center"
-                                    colspan="3"
-                                    data-translate="sms_options_response_tracking_empty">
-                                </td>
-                            </tr>
-                        </tbody>
-                    </table>
-                </div>
+                <oui-datagrid class="reduce-padding" rows="TelecomSmsOptionsResponseCtrl.smsResponse.trackingOptions">
+                    <oui-column title="'sms_options_response_tracking_media_label' | translate" property="media" sortable>
```
Comment (neolitec) [inline client/app/telecom/sms/options/response/telecom-sms-options-response.html]: According to CX guidelines, all grid should have a default sorted column. In most cases we'll have to add `sortable="asc"` on the first column.

File: client/app/telecom/sms/options/response/telecom-sms-options-response.html [hunk 4]
```diff
@@ -175,99 +175,41 @@ <h1 data-translate="sms_options_response"></h1>
...
+                    <oui-column property="target">
+                        <strong class="d-block"
+                                data-ng-bind="$row.target">
+                        </strong>
+                        <span data-translate="sms_options_response_tracking_email_sender_label"
+                            data-translate-values="{ sender: $row.sender }">
+                        </span>
+                    </oui-column>
+
+                    <oui-action-menu align="end" compact>
```
Comment (neolitec) [inline client/app/telecom/sms/options/response/telecom-sms-options-response.html]: A recent discussion about `data-` prefixes lead us to keep all `data-` on custom attributes whenever we are in the managers. => `data-align`, `data-compact`, same thing for the datagrid attributes especially for `title` but except for `id`, `name`, etc...

File: client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js [hunk 5]
```diff
@@ -78,12 +83,18 @@ angular.module("managerApp").controller("TelecomSmsSmsTemplatesCtrl", class Tele
...
+                this.getDetails(model).then((template) => {
+                    this.templates.paginated.push(template);
```
Comment (neolitec) [inline client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.controller.js]: You should not have to deal with pagination while using oui-datagrid...
You should pass `templates.raw` to datagrid as `row` attribute. And then, for each line loading, you can use `row-loader`: a function that you define in the controller, taking `$row` as parameter and returns a complete line (with all columns) wrapped in a promise. In so doing, `getDetails(model)` is only run for displayed rows.

File: client/app/telecom/sms/options/response/telecom-sms-options-response.html [hunk 6]
```diff
@@ -176,99 +176,41 @@ <h1 data-translate="sms_options_response"></h1>
...
-                        </tbody>
-                    </table>
-                </div>
+                <oui-datagrid class="reduce-padding" data-rows="TelecomSmsOptionsResponseCtrl.smsResponse.trackingOptions">
+                    <oui-column data-title="'sms_options_response_tracking_media_label' | translate" data-property="media" data-sortable="asc">
+                        <span class="text-capitalize visible-xs"
+                              data-ng-bind="$row.media">
+                        </span>
+                        <i class="ovh-font ovh-font-1x4 ml-2 hidden-xs"
+                            data-ng-attr-title="{{ $row.media }}"
```
Comment (Jisay) [inline client/app/telecom/sms/options/response/telecom-sms-options-response.html]: one space too much ;-)

File: client/app/telecom/sms/options/response/telecom-sms-options-response.html [hunk 7]
```diff
@@ -176,99 +176,41 @@ <h1 data-translate="sms_options_response"></h1>
...
+                            }">
+                        </i>
+                    </oui-column>
+
+                    <oui-column data-property="target">
+                        <strong class="d-block"
+                                data-ng-bind="$row.target">
+                        </strong>
+                        <span data-translate="sms_options_response_tracking_email_sender_label"
+                            data-translate-values="{ sender: $row.sender }">
```
Comment (Jisay) [inline client/app/telecom/sms/options/response/telecom-sms-options-response.html]: small problem of indentation :-)

File: client/app/telecom/sms/sms/hlr/telecom-sms-sms-hlr.html [hunk 8]
```diff
@@ -140,224 +140,58 @@ <h1 data-translate="sms_sms_hlr_title"></h1>
...
+
+                    <oui-column data-title="'sms_sms_hlr_read_label_cost' | translate" data-property="status">
+                        <span data-translate="sms_sms_hlr_read_cost_value"
+                              data-translate-values="{ credit: $row.status === 'done' ? 0.1 : 0 }">
+                        </span>
+                    </oui-column>
+
+                    <oui-column data-title="'sms_sms_hlr_read_label_status' | translate" data-property="status">
+                        <i class="ovh-font"
+                            data-ng-class="{
```
Comment (Jisay) [inline client/app/telecom/sms/sms/hlr/telecom-sms-sms-hlr.html]: indentation

File: client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.html [hunk 9]
```diff
@@ -94,211 +94,56 @@ <h1 data-translate="sms_sms_templates_title"></h1>
...
+                {{$row.description || '-'}}
+            </oui-column>
+
+            <oui-column data-title="'sms_sms_templates_list_title_message' | translate" data-property="message" data-sortable>
+                {{$row.message || '-'}}
+            </oui-column>
+
+            <oui-column data-title="'sms_sms_templates_list_title_status' | translate" data-property="status" data-sortable>
+                <i class="ovh-font"
+                    data-ng-class="{
```
Comment (Jisay) [inline client/app/telecom/sms/sms/templates/telecom-sms-sms-templates.html]: one space too much here too (hihi ;-) )

File: client/app/telecom/telecom.less [hunk 10]
```diff
@@ -28,5 +28,9 @@
             margin: 0;
         }
     }
+}
 
+oui-datagrid[class^="reduce-padding"] td.oui-datagrid__cell {
```
Comment (Jisay) [inline client/app/telecom/telecom.less]: do we really want these styles ?

**Pull request merged**